incubator/hnc/internal/reconcilers/hierarchy_config.go

```diff
@@ -398,6 +404,9 @@ func (r *HierarchyConfigReconciler) syncLabel(log logr.Logger, nsInst *corev1.Na
 		anc = anc.Parent()
 		depth++
 	}
+
+	labels[propagatePrefix+nsInst.Name] = "true"
+	ns.SetLabels(labels)
```

Comment:
This doesn't look correct to me. It looks like we're _only_ storing the labels that HNC itself has created, but we should be storing _all_ the labels on the namespace, including the ones set by the users.

Reply:
Done